### ISSUE TITLE
Implementation of Freeze Protection mode

### ISSUE THREAD

**Issue by HausnerR** [mac-zhou/midea-ac-py#190]
Hey, I saw in source code (especially in midea-msmart repo) comments about freeze protection mode. Any chance to implement it based on this (https://github.com/mac-zhou/midea-msmart/blob/97723d055d8e5a3775eaaca96807f2dbbd6db2aa/msmart/device/AC/command.py#L478)?

Best,
HausnerR

**Comment by mill1000** [comment]
We need a little more info on how to control the freeze protection mode. The comment referenced above only has information on what bit to check to see if it's enabled or not. We'd need to know what bit to send to control the mode.

Some packet captures between the mobile app and AC device would be a good start.

**Comment by audreez** [comment]
i can probably provide that, is it not encrypted?

**Comment by audreez** [comment]
i captured this between the minisplit and the router when i sent the FP command, is that what you are looking for ?

![image](https://user-images.githubusercontent.com/12138529/186285271-07a10e6e-fd67-4d55-a034-16845bbf3e80.png)

**Comment by mill1000** [comment]
No, I believe we need to packets between your phone and the minisplit unit. Those are more difficult to capture.

**Comment by HausnerR** [comment]
Hey, I found that this repo probably have code for enabling/disabling FP mode (based on readme).

https://github.com/reneklootwijk/midea-uart

Maybe it can be analyzed and ported into this project?

**Comment by mill1000** [comment]
Nice find. Working on this functionality in a branch [here](https://github.com/mill1000/midea-ac-py/tree/feature/freeze_protection).

**Comment by audreez** [comment]
want me to test?

**Comment by mill1000** [comment]
That would be great. You'll need to manually install this branch of msmart as well: https://github.com/mill1000/midea-msmart/tree/feature/freeze_protection

**Comment by audreez** [comment]
getting this when trying to add the repo <Integration mill1000/midea-msmart> Repository structure for master is not compliant

**Comment by mill1000** [comment]
midea-msmart isn't a HACS repo. It has to be installed manually with `pip` in your HA environment

**Comment by mill1000** [comment]
I've updated the branches and I think this functionality is ready, but I can't actually test as my units don't implement this.

Manually install msmart branch in your HA environment
```
pip install git+https://github.com/mill1000/midea-msmart.git@feature/freeze_protection
```

Manually install midea-ac-py branch in your HA custom_components
```
git clone -b feature/freeze_protection https://github.com/mill1000/midea-ac-py.git
cp -r midea-ac-py/custom_components/midea_ac/ <path-to-custom-components>
```

**Comment by audreez** [comment]
thanks! im planning on testing this weekend

**Comment by audreez** [comment]
getting this after copying it to custom components 


Logger: homeassistant.components.climate
Source: helpers/entity_platform.py:193
Integration: Climate ([documentation](https://www.home-assistant.io/integrations/climate), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+climate%22))
First occurred: 2:10:53 PM (2 occurrences)
Last logged: 2:10:53 PM

The midea_ac platform for the climate integration does not support platform setup. Please remove it from your config.

**Comment by mill1000** [comment]
Try removing any `midea_ac` sections from your `configuration.yaml`. My branch uses the GUI for configuration.

**Comment by audreez** [comment]
got it working, how/where do i turn on the FP ? usually it needs to be in heat operation and you set the FP preset/mode but i dont see anything in operation or presets

**Comment by mill1000** [comment]
Ah. I added it as a switch. You should see it on the device page if the unit report supporting it.

It appears I may have misunderstood this function. Would it be better suited as a preset? I see there's a PRESET_AWAY would that be equivalent?

**Comment by audreez** [comment]
added both of my splits (both support FP) to it but no switch was created, only 2 temp sensors and climate

yes away kind of makes sense

**Comment by audreez** [comment]
yes them more i think, preset "away" would be great, only available in heat mode as you can not turn on FP in any other mode

**Comment by mill1000** [comment]
Can you enable logging? Add the following to your YAML.
```yamllogger:
  default: warning
  logs:
    custom_components.midea_ac: info
    msmart: debug
```

Then check your `home-assistant.log`, you may see something like this. Post yours here.
```
2022-09-10 10:03:53 INFO (MainThread) [custom_components.midea_ac] Querying device capabilities.
2022-09-10 10:03:54 INFO (MainThread) [custom_components.midea_ac.sensor] Setting up sensor platform.
2022-09-10 10:03:55 INFO (MainThread) [custom_components.midea_ac.binary_sensor] Setting up binary sensor platform.
2022-09-10 10:03:55 INFO (MainThread) [custom_components.midea_ac.switch] Setting up switch platform.
2022-09-10 10:03:55 INFO (MainThread) [custom_components.midea_ac.switch] Device does not support freeze protection.
2022-09-10 10:03:55 INFO (MainThread) [custom_components.midea_ac.climate] Setting up climate platform.
```

You may also see something like
```
2022-09-10 21:51:33 DEBUG (SyncWorker_0) [msmart.device.AC.command] Supported capabilities: {'eco_mode': False, 'eco_mode_2': True, 'freeze_protection': False, 'heat_mode': False, 'cool_mode': True, 'dry_mode': True, 'auto_mode': True, 'swing_horizontal': False, 'swing_vertical': True, 'nest_check': True, 'nest_need_change': False, 'turbo_heat': False, 'turbo_cool': True, 'fan_speed_control': False, 'cool_min_temperature': 16.0, 'cool_max_temperature': 30.0, 'auto_min_temperature': 16.0, 'auto_max_temperature': 30.0, 'heat_min_temperature': 16.0, 'heat_max_temperature': 30.0, 'decimals': True, 'light_control': True}
```
which we can confirm FP is supported.

**Comment by audreez** [comment]
2022-09-10 22:06:50.970 WARNING (MainThread) [custom_components.midea_ac.helpers] Device does not support 'get_capabilities' method.
2022-09-10 22:06:51.085 WARNING (MainThread) [custom_components.midea_ac.helpers] Device does not support 'filter_alert' property.
2022-09-10 22:06:51.725 WARNING (MainThread) [custom_components.midea_ac.helpers] Device does not support 'toggle_display' method.
2022-09-10 22:06:51.727 WARNING (MainThread) [custom_components.midea_ac.helpers] Device does not support 'supports_freeze_protection' property.

**Comment by mill1000** [comment]
Looks like you're running an older version of msmart. If you're running a docker image, be sure to run the above `pip install` command within the container.

IF you're running HAOS you can use this extension, https://github.com/hassio-addons/addon-ssh, to access the shell and run pip

**Comment by audreez** [comment]
i run haos on proxmox, at some point i reinstalled haos and loaded from a backup so msmart was gone
haos didnt have pip, i installed python + pip and installed msmart, 
now i can run midea-discover -i ip_address and it shows all the keys etc, (from ssh terminal addon)

when i restart home assistant log still displays same "not supported" messages any other ideas?

**Comment by audreez** [comment]
maybe dont require feature check ? jsut add an option: 
![image](https://user-images.githubusercontent.com/12138529/189514547-0ca82d75-26c5-4dd1-b6f1-97e0a9b0fb2b.png)

**Comment by audreez** [comment]
well deleted everything and reinstalled, still same issue
![image](https://user-images.githubusercontent.com/12138529/189515037-82d8ff6d-d582-4a5e-82af-c1c45a9c09c3.png)

![image](https://user-images.githubusercontent.com/12138529/189515049-d5ce2d61-42c1-4f53-ab48-4dd2cef9dc27.png)

**Comment by mill1000** [comment]
> i run haos on proxmox, at some point i reinstalled haos and loaded from a backup so msmart was gone haos didnt have pip, i installed python + pip and installed msmart, now i can run midea-discover -i ip_address and it shows all the keys etc, (from ssh terminal addon)
> 
> when i restart home assistant log still displays same "not supported" messages any other ideas?

HomeAssistant is still run within a container on HAOS. You most likely weren't inside the container (hence why you had to install Python)

The feature check is there to provide backwards compatibility with older msmart versions. Without it the integration would completely fail.

**Comment by audreez** [comment]
you are correct, i assumed the ssh addon takes you inside the homeassistnat docker container i was wrong,

anyway i got inside the docker container and installed msmart now i see freeze protection

![image](https://user-images.githubusercontent.com/12138529/189539222-17938541-28a4-45d9-9454-c7a98af67abd.png)

so the switch does turn on the freeze protection, but it doesn't stay "on" to display status that its actually on and it does not turn it off if i try to toggle it again

**Comment by mill1000** [comment]
Ok. That's some progress. Were there any log messages when trying to turn off the freeze protection? What model is your unit? I'm trying to understand how FP is used in the app/unit.

**Comment by audreez** [comment]
no errors in the log
FP is used as vacation mode / away mode basically if you are leaving the house for the weekend you turn on heat mode and the FP which basically will keep your house from freezing and will set itself to i think ~46-47F  (on regular remote lowest it goes is 62F i believe) 
on the app itself the heating set point stays the same you just torn on FP and it ignores the setpoint and uses 46F (hard set, non adjustable) as baseline, screen on the mini split also shows FP instead of temperature

here's a pic of the app

![image](https://user-images.githubusercontent.com/12138529/189553333-350f9d1e-4c97-49f2-a8af-9e7bc0071c9c.png)

**Comment by mill1000** [comment]
Ok thanks. That's helpful. Is FP mutual exclusive with ECO and Boost?

Can you enable debug logs when trying to turn on and off the FP? Even better if you can capture logs while using the app to toggle FP mode. We should be able to confirm that I'm using the right bits.
```
  default: warning
  logs:
    msmart: debug
```

**Comment by mill1000** [comment]
I made another branch that re-implements FP as an "Away" preset. Make sure to grab the latest msmart branch as well.
https://github.com/mill1000/midea-ac-py/tree/feature/fp_away_mode_preset

**Comment by audreez** [comment]
im goign to setup logs and run some tests, in FP mode i can turn on Boost, but not ECO


so i ran:
pip install git+https://github.com/mill1000/midea-msmart.git@feature/freeze_protection --upgrade 
then:
git clone -b feature/fp_away_mode_preset https://github.com/mill1000/midea-ac-py.git
then: 
cp -r midea-ac-py/custom_components/midea_ac/ /config/custom_components/
then restarted

switched are now not working at all (expected) but i don't see an away preset 
![image](https://user-images.githubusercontent.com/12138529/190200074-585860a6-fff6-4c6d-8fdb-7151d655f895.png)

**Comment by audreez** [comment]
for the logging, are you looking for these when i turn on/off FP? there is a lot of fluff there just narrowing down on what you need would help

	Line 20: 2022-09-14 09:55:35.245 DEBUG (SyncWorker_4) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 33: 2022-09-14 09:55:50.981 DEBUG (SyncWorker_4) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 46: 2022-09-14 09:56:08.983 DEBUG (SyncWorker_6) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 55: 2022-09-14 09:56:20.985 DEBUG (SyncWorker_7) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 64: 2022-09-14 09:56:35.988 DEBUG (SyncWorker_2) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 81: 2022-09-14 09:57:05.985 DEBUG (SyncWorker_6) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000000000
	Line 90: 2022-09-14 09:57:20.984 DEBUG (SyncWorker_0) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 99: 2022-09-14 09:57:35.987 DEBUG (SyncWorker_1) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584b0000000000000000800000
	Line 112: 2022-09-14 09:57:53.995 DEBUG (SyncWorker_4) [msmart.device.AC.command] State response payload: c00185667f7f003c800004584a0000000000000000800000

**Comment by mill1000** [comment]
Could you post the full log? I wonder if the capabilities aren't reporting FP support.

EDIT: Could also be that I'm checking the wrong operation mode when trying to add the Away preset

**Comment by audreez** [comment]
2022-09-14 10:50:16.332 DEBUG (SyncWorker_3) [msmart.lan] Got responses len: 150
2022-09-14 10:50:16.333 DEBUG (SyncWorker_3) [msmart.device.base] Got responses from xxx.xxx.xxx.31:6444 Version: 3 Count: 1 Spend time: 0.62
2022-09-14 10:50:16.333 DEBUG (SyncWorker_3) [msmart.device.base] Update from xxx.xxx.xxx.31:6444 aa29ac00000000000303b5071202010113020101140201011502010116020100170201001a0201017a40
2022-09-14 10:50:16.333 DEBUG (SyncWorker_3) [msmart.device.AC.command] Capabilities response payload: b5071202010113020101140201011502010116020100170201001a020101
2022-09-14 10:50:16.334 DEBUG (SyncWorker_3) [msmart.device.AC.command] Supported capabilities: {'eco_mode': True, 'eco_mode_2': False, 'freeze_protection': True, 'heat_mode': True, 'cool_mode': False, 'dry_mode': True, 'auto_mode': True, 'swing_horizontal': True, 'swing_vertical': True, 'power_cal': False, 'power_cal_setting': False, 'nest_check': False, 'nest_need_change': False, 'turbo_heat': True, 'turbo_cool': True}
2022-09-14 10:50:21.531 DEBUG (SyncWorker_8) [msmart.base_command] Frame data: aa21ac8d000000000003418100ff03ff000200000000000000000000000003028b4e
2022-09-14 10:50:21.535 DEBUG (SyncWorker_8) [msmart.device.base] pkt_builder: xxx.xxx.xxx.31:6444 len: 104 data: 5a5a011168002000000000003515320a0e0916149c780000001e0000000000000000000000000000b8436dd15e84d5a4fc6fbf77d2b1248672946460bceafc6eae86e4200896522a734e1faa37ce959a047497632b36c8a8077a712db26ef65984ec21b08dbd600a
2022-09-14 10:50:21.537 DEBUG (SyncWorker_8) [msmart.lan] Socket xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 7.5 tcp_key: xxxxx
2022-09-14 10:50:21.537 DEBUG (SyncWorker_8) [msmart.lan] Sending xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 7.5 message: 8370008e2066a192fe3cdac7b33a292d94b17e45982e3e7d51553987a47de90c3334186056c55d5394fd554726cfa8b2ff505af2696341b6e444b735b84cce356116320f75567cd85a1940fd2f68d4cc35103b0b8e7612a0308293cf55c18374bc55d285a1cfc46e7aa45ff94b9238741b54c017b051f85c3d7096fecb98aa2a50e96e1cc8ceb3b025a014fdef952869a5279545273a
2022-09-14 10:51:08.117 DEBUG (SyncWorker_1) [msmart.lan] Recv xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 54.08 Response: 8370008e2063fe9bef213a5997f4f79e24ae97182df9e0ebd6c8beb9c739f67a3e61bfe1d871467a880ab9cd7cc9c7946184e4972a9bc7c9f51871e82c0ed5f193dbe4f59d362d19100c73643b7f315cc78035102a9f2acf5a54757e5e072aef00a15a8901282f1b59b98e74a1250321fe66c55272a9bd42ef82ca30b1a92dbbfd20ce2df4185d98abfcd55e774ee2f12a52fee47bb4
2022-09-14 10:51:08.117 DEBUG (SyncWorker_1) [msmart.lan] Got responses len: 150
2022-09-14 10:51:08.118 DEBUG (SyncWorker_1) [msmart.device.base] Got responses from xxx.xxx.xxx.31:6444 Version: 3 Count: 1 Spend time: 0.73
2022-09-14 10:51:08.118 DEBUG (SyncWorker_1) [msmart.device.base] Update from xxx.xxx.xxx.31:6444 aa23ac00000000000303c00185667f7f003c8000045b49000000000000000000000053ca
2022-09-14 10:51:08.118 DEBUG (SyncWorker_1) [msmart.device.AC.command] State response payload: c00185667f7f003c8000045b490000000000000000000000
2022-09-14 10:51:22.392 DEBUG (SyncWorker_6) [msmart.base_command] Frame data: aa21ac8d000000000003418100ff03ff00020000000000000000000000000306eaeb
2022-09-14 10:51:22.393 DEBUG (SyncWorker_6) [msmart.device.base] pkt_builder: xxx.xxx.xxx.31:6444 len: 104 data: 5a5a011168002000000000002716330a0e0916149c780000001e0000000000000000000000000000b8436dd15e84d5a4fc6fbf77d2b12486062fc0a1cd7d96dbe819b07b3b0e41b807461b25fbde92e23b151c4a5df712fdfd62c8e0becc19ea22643a2eead1ab32
2022-09-14 10:51:22.393 DEBUG (SyncWorker_6) [msmart.lan] Socket xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 68.35 tcp_key: xxxxx
2022-09-14 10:51:22.393 DEBUG (SyncWorker_6) [msmart.lan] Sending xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 68.35 message: 8370008e20661d3f76b0826c34cd8b14061a7e3fe42542d51565e8aeb1b287d98cfe3f84efe41945d2c86c3d71e04d8355252cc88727d924e82c8ed96adea86a93d4f29d89c399acfe0492d163f44dcfde84ae5621614a39524721c596e32919639416d4bd9da9fa5aed7f25a70d3dab1637977138a8d96d91a707f5729ccbb5693d97681e20c0aece3b44478f11d0a24973ef33c28d
2022-09-14 10:51:23.115 DEBUG (SyncWorker_6) [msmart.lan] Recv xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 69.07 Response: 8370008e206363c5fcefe92634ca4671f2f844c43d39008e65460d65fc220839d8e1c1a3c66fe4158c9798505dafd212e1dd1560587141ff4618da7cda6f35076a1dd4d386f8699e7eb2c27804ce3a6fd0526596f2e52a1f3e93aea0addcf00881866ecd3a7fcd57e6881c52fec393d7698a69c1ef791e6ce78a5b926d0b5836e86c4c8dd8ac22e039820b8eeba3c37bb27a916c56ba
2022-09-14 10:51:23.116 DEBUG (SyncWorker_6) [msmart.lan] Got responses len: 150
2022-09-14 10:51:23.116 DEBUG (SyncWorker_6) [msmart.device.base] Got responses from xxx.xxx.xxx.31:6444 Version: 3 Count: 1 Spend time: 0.72
2022-09-14 10:51:23.117 DEBUG (SyncWorker_6) [msmart.device.base] Update from xxx.xxx.xxx.31:6444 aa23ac00000000000303c00185667f7f003c8000045b49000000000000000000000053ca
2022-09-14 10:51:23.117 DEBUG (SyncWorker_6) [msmart.device.AC.command] State response payload: c00185667f7f003c8000045b490000000000000000000000
2022-09-14 10:51:37.393 DEBUG (SyncWorker_1) [msmart.base_command] Frame data: aa21ac8d000000000003418100ff03ff00020000000000000000000000000307b420
2022-09-14 10:51:37.394 DEBUG (SyncWorker_1) [msmart.device.base] pkt_builder: xxx.xxx.xxx.31:6444 len: 104 data: 5a5a011168002000000000002725330a0e0916149c780000001e0000000000000000000000000000b8436dd15e84d5a4fc6fbf77d2b124866ecff7bf9e9fe8567b674002630ed398ccd2da10984b3e911e7b18c141e9cc0c68f2db9fd253b2eb848b35490fec9d84
2022-09-14 10:51:37.394 DEBUG (SyncWorker_1) [msmart.lan] Socket xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 83.35 tcp_key: xxxxx
2022-09-14 10:51:37.394 DEBUG (SyncWorker_1) [msmart.lan] Sending xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 83.35 message: 8370008e20662667e9455d603c25f86a835903d58af4eb9fc87e2a8372bd1d1e9cd9b958084508ace3fe9304e55e0bbafcedbaeaad222f0d62142fc5b3adc0d56e6f2b55fd22392a7f0018119feaf055a11a06954865b6c91471580d8f557f79c8ea7f7f3d448235f543b19b97898493fc2ff65c6bce3db3ac04b710258dcf5a6dac0265b59306e513fc47ab987364a189b07dc151df
2022-09-14 10:51:38.117 DEBUG (SyncWorker_1) [msmart.lan] Recv xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 84.08 Response: 8370008e206397916d9a7d89b65b8eb484772167a7f5950b8285e74f2da3ba3f9cda6e10d5974957fd72fc172fa08746ddbb136edfd27633a6344d69bd97968570f1bb92883e1b9ceeaaca876a0403bf6c1e27569a596eae5418228d1bb2181a1056a30d85d2fc7f4cb12d050490268d0d62de030918711e5e88acc30bd7e15a5b03f48dc154724b14add5dff8400c7e8003dbe69bed
2022-09-14 10:51:38.118 DEBUG (SyncWorker_1) [msmart.lan] Got responses len: 150
2022-09-14 10:51:38.118 DEBUG (SyncWorker_1) [msmart.device.base] Got responses from xxx.xxx.xxx.31:6444 Version: 3 Count: 1 Spend time: 0.72
2022-09-14 10:51:38.118 DEBUG (SyncWorker_1) [msmart.device.base] Update from xxx.xxx.xxx.31:6444 aa23ac00000000000303c00185667f7f003c8000045b49000000000000000000000053ca
2022-09-14 10:51:38.118 DEBUG (SyncWorker_1) [msmart.device.AC.command] State response payload: c00185667f7f003c8000045b490000000000000000000000
2022-09-14 10:51:52.394 DEBUG (SyncWorker_6) [msmart.base_command] Frame data: aa21ac8d000000000003418100ff03ff00020000000000000000000000000308f5de
2022-09-14 10:51:52.395 DEBUG (SyncWorker_6) [msmart.device.base] pkt_builder: xxx.xxx.xxx.31:6444 len: 104 data: 5a5a011168002000000000002734330a0e0916149c780000001e0000000000000000000000000000b8436dd15e84d5a4fc6fbf77d2b124862de3ee43a85a38d3e77ee80be0616b858fa1228a4d31ca8d78221616c9c111ef737cfcd907afc400283466bac248e9c3
2022-09-14 10:51:52.395 DEBUG (SyncWorker_6) [msmart.lan] Socket xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 98.35 tcp_key: xxxxx
2022-09-14 10:51:52.395 DEBUG (SyncWorker_6) [msmart.lan] Sending xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 98.35 message: 8370008e2066a288eca30ddb596ae4d3ae01179bfe2e1510229964e285c2e41f75201d8090c8153d24c0811e7d68a7a46cc0a9a1ba11e5cd9c1b058a89665db850f18601cae03fae4004ac4257b6a1a9b5d527761aba5c90abf0b6247d3b19c869e8c82a48732df271bb7928649ee02a34862211cfd5300dfe50814ded9adfd6933bd00279ba4d0a3dc073b99508fb8f215155f175cb
2022-09-14 10:51:53.123 DEBUG (SyncWorker_6) [msmart.lan] Recv xxx.xxx.xxx.15:36714 -> xxx.xxx.xxx.31:6444 retries: 0 time: 99.08 Response: 8370008e20638e900b76af3b60e361ddf28e37ac4c964a4b4eb13dea3b44c70aff1e21f4ea70a42d781485d67eb3949e247567984892c2d84a52d6797bb907b61f8366f825d14db5616bb92ac7a71355bdee87228b62907e9666647b7f33d581a94fc6f291d8d71e40ec8ae39ad89ef4528d43c6848ac6d02fb698bc8895b54c82e2529f8bde3d00d3190cc708835f949a6a6ecba303
2022-09-14 10:51:53.123 DEBUG (SyncWorker_6) [msmart.lan] Got responses len: 150
2022-09-14 10:51:53.123 DEBUG (SyncWorker_6) [msmart.device.base] Got responses from xxx.xxx.xxx.31:6444 Version: 3 Count: 1 Spend time: 0.73
2022-09-14 10:51:53.124 DEBUG (SyncWorker_6) [msmart.device.base] Update from xxx.xxx.xxx.31:6444 aa23ac00000000000303c00185667f7f003c8000045b49000000000000000000000053ca
2022-09-14 10:51:53.124 DEBUG (SyncWorker_6) [msmart.device.AC.command] State response payload: c00185667f7f003c8000045b490000000000000000000000

**Comment by audreez** [comment]
redacted sens infoa bit and attached
[midealog.txt](https://github.com/mac-zhou/midea-ac-py/files/9568369/midealog.txt)

**Comment by mill1000** [comment]
Thanks. Looks like FP is supported. Could you remove and re-install msmart via pip? I just want to make sure it's using the latest version of that branch and I'm not sure how `--upgrade` works when installing from git

**Comment by audreez** [comment]
she werks! it does turn on and off FP on the split
![image](https://user-images.githubusercontent.com/12138529/190219403-d96e3e30-fc5f-4804-880d-2cc46985680b.png)
![image](https://user-images.githubusercontent.com/12138529/190219763-73c60ee9-eb15-422d-8d2f-492513dc7c94.png)

**Comment by mill1000** [comment]
Awesome! Does it properly detect Away mode if you use the remote to enable it?

**Comment by audreez** [comment]
i used the app to run in and off (don't us the remote dnt know where it is) and it does detect it both ways

**Comment by audreez** [comment]
are you going to merge tp main branch or should i keep using your version?
also let me buy you a beer!

**Comment by mill1000** [comment]
I'll merge the FP changes into my "future" branch and create PRs for the main branch as well.
